### PR TITLE
Configure GitLab CI pipeline

### PR DESCRIPTION
**Pull request by antti-mikael**

### CODE REVIEW

**Comment by antti-mikael** [comment]
This PR is based on #94.

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  echo "Database is up!"
+fi
+
+# Apply database migrations
+if [[ "$APPLY_MIGRATIONS" = "1" ]]; then
+    echo "Applying database migrations..."
+    ./manage.py migrate --noinput
+fi
+
+if [[ "$SETUP_DEV_OIDC" = "1" ]]; then
```
Comment (ruriky) [inline docker-entrypoint.sh]: With this PR, the code blocks:
- SETUP_DEV_OIDC
- GENERATE_OPENID_KEY
- CREATE_SUPERUSER
.. are not executed. Should they be executed?

File: Dockerfile [hunk 2]
```diff
@@ -1,16 +1,88 @@
...
+      libxmlsec1-dev \
+      libxml2-dev \
+      netcat \
+      pkg-config \
+    && pip install -U pip \
+    && pip install --no-cache-dir  -r /app/requirements.txt \
+    && pip install --no-cache-dir  -r /app/requirements-prod.txt \
+    && apt-cleanup.sh build-essential pkg-config git
+
+COPY docker-entrypoint.sh /app
```
Comment (ruriky) [inline Dockerfile]: No "--chown=appuser:appuser" set. Does the appuser user have sufficient execution rights?

File: Dockerfile [hunk 3]
```diff
@@ -1,16 +1,88 @@
...
+# is required for compiling SCSS files.
+# =========================================================
 
-ENV PYTHONUNBUFFERED 0
+RUN apt-install.sh \
+      libxmlsec1-dev \
+      libxml2-dev \
+      pkg-config \
+      git \
+      curl \
```
[the remark this answers is not in the document]
Comment (antti-mikael) [reply Dockerfile]: Out of scope.

File: docker-entrypoint.sh (hunk 1 above)
Comment (antti-mikael) [reply docker-entrypoint.sh]: I'm unsure about this but it is something that can be added later if needed.

File: Dockerfile (hunk 2 above)
Comment (antti-mikael) [reply Dockerfile]: Out of scope.

File: .prod/uwsgi.ini [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[uwsgi]
+http-socket = :8000
+chdir = /app
+module = tunnistamo.wsgi
+static-map = /media=/app/media
+static-map = /sso/static=/var/tunnistamo/static
+uid = appuser
```
[the remark this answers is not in the document]
Comment (antti-mikael) [reply .prod/uwsgi.ini]: Out of scope.

**Pull request merged**